sdk/storage/azblob/highlevel.go

```diff
@@ -0,0 +1,646 @@
+package azblob
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"io"
+	"net/http"
+	"sync"
+
+	"bytes"
+	"errors"
+	"os"
+	//"github.com/Azure/azure-pipeline-go/pipeline"
+)
+
+//// CommonResponse returns the headers common to all blob REST API responses.
+type CommonResponse interface {
+	//// ETag returns the value for header ETag.
+	//ETag() *string
+	//
+	//// LastModified returns the value for header Last-Modified.
+	//LastModified() time.Time
+	//
+	//// RequestID returns the value for header x-ms-request-id.
+	//RequestID() string
+
+	//// Date returns the value for header Date.
+	//Date() time.Time
+	//
+	//// Version returns the value for header x-ms-version.
+	//Version() string
+
+	// Response returns the raw HTTP response object.
+	Response() *http.Response
+}
+
+// HighLevelUploadToBlockBlobOption identifies options used by the UploadBufferToBlockBlob and UploadFileToBlockBlob functions.
+type HighLevelUploadToBlockBlobOption struct {
```

Comment:
To follow up later: the keyword "HighLevel" seems a bit off, ideally these struct names should align with func that takes them in.